apps/backend/src/auth/login.guard.ts

```diff
@@ -0,0 +1,12 @@
+import { ExecutionContext, Injectable } from '@nestjs/common';
+import { AuthGuard } from '@nestjs/passport';
+
+@Injectable()
+export class LoginGuard extends AuthGuard('oidc') {
+  async canActivate(context: ExecutionContext) {
+    const result = (await super.canActivate(context)) as boolean;
+    const request = context.switchToHttp().getRequest();
```

Comment:
![](https://media.giphy.com/media/1oHqrKnyu2vhWJDPNc/giphy.gif)

Reply:
![](https://tenor.com/view/neither-do-i-same-me-too-mutual-me-neither-gif-15492197)

Reply:
![](https://media.tenor.com/KMA3m6dK4IgAAAAd/neither-do-i-same.gif)

Reply:
nice

Reply:
Ok, I know why we change context to HTTP. In [documentation](https://docs.nestjs.com/fundamentals/execution-context#current-application-context) I've found this and it's connected to multiple communication ways in NestJS (REST/gRCP/GraphQL).